src/main/java/org/jenkins/plugins/lockableresources/LockableResourcesManager.java

```diff
@@ -656,31 +677,16 @@ public void unlockNames(
         synchronized (this.syncResources) {
             this.freeResources(this.fromNames(resourceNamesToUnLock), build);
 
-            // process as many contexts as possible
-            this.lastCheckedQueueIndex = -1;
-            while (resourceNamesToUnLock.size() > 0 && proceedNextContext(inversePrecedence)) {
-                for (String resourceName : resourceNamesToUnLock) {
-                    LockableResource r = fromName(resourceName);
-                    if (r == null) {
-                        // probably it was ephemeral resource and does not exists now
-                        // therefore we need to check the whole queue later
-                        break;
-                    }
-                    if (!r.isFree()) {
-                        // i sno more free, that means, we does not need to check it in the queue now
-                        resourceNamesToUnLock.remove(resourceName);
-                        break;
-                    }
-                }
+            while (proceedNextContext()) {
```

Comment:
Empty body? Seems wasteful.

Reply:
I need the while loop to proceed all possible geued items.  So the body will be always empty.